woocommerce/woocommerce-google-analytics-integration
# Allow more precise consent mode customizations

### Changes proposed in this Pull Request:



Allow more precise consent mode customizations. Use the filtered array with potential customizations just after `gtag` is defied, so the consent mode is set up before it's used, allowing cookies to be created correctly for recurring `granted` visitors.

### Checks:


### Screenshots:




### Detailed test instructions:


1. Build, Install and activate the extension
4. Open [Tag Assistant](https://tagassistant.google.com/) 
5. Open shop page the page 
6. In Tag assistant you should see the "Consent" as the first entry.
7. If you're connecting from EEA region, you should see "Consent" being Denied by default
   ![image](https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/17435/361077b9-7e43-4edb-9107-2fad4ca9deb8)
8. If you're connecting from non-EEA region, you should see "Consent" not set 
   ![image](https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/17435/ad61fcf8-4377-4218-a4af-721047c24af2)
10. Check the Readme description 
11. Add the snippet to customize the consent mode:
```php
add_filter( 'woocommerce_ga_gtag_consent_modes', function ( $consent_modes ) {
    $consent_modes[] =
		array(
            'analytics_storage' => 'granted',
            'region'            => array( 'PL' ), // <------ your region
        );
   return $consent_modes;
} );
```
12. Check that the second Consent entry is added with custom defaults before the container is loaded
   ![image](https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/17435/c6601ea2-f36c-44c3-ad40-7fca122f2f12)
13. Check that the cookies were created
   ```js
   (await cookieStore.getAll()).filter((c)=>c.name.includes('ga'))
   ```
   ![image](https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/17435/039d19fb-974a-4866-9b08-4aaf14a7666e)




### Additional details:


### Changelog entry
_no changelog entry, as this is a part of the new to-be-released feature_
